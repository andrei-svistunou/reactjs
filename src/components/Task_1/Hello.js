import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Hello extends Component {
  render() {
    return <h3>Hello, {this.props.name}</h3>;
  }
}

Hello.propTypes = {
  name: PropTypes.string.isRequired
};

export default Hello;
