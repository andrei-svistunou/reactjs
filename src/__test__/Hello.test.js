import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Hello from '../components/Task_1/Hello';

describe('<Hello />', () => {
  test('should be render a default Hello', () => {
    const hello = shallow(<Hello name='test' />);

    expect(toJson(hello)).toMatchSnapshot();
  });

  test('should be get props', () => {
    const hello = mount(<Hello name='Jest' />);

    expect(hello.prop('name')).toEqual('Jest');
  });
});
