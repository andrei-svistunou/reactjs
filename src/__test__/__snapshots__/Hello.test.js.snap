// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hello /> should be render a default Hello 1`] = `
<h3>
  Hello, 
  test
</h3>
`;
